### PR TITLE
Backend image resizing

### PR DESCRIPTION
**Pull request by rebbrunner**
Merge before PR to refactor code - models changes to use helper functions.
Converted tabs to space.
Line endings converted from old fork in views.py.  Lines that changed 108-114 added a check for max file size.

### CODE REVIEW

**Comment by rebbrunner** [comment]
Add backend image resizing
Bug fix cropping
Add max file size
Remove crop
Are the changes

Everything else is merging branches into this one to resolve merge conflicts before they happen.

**Comment by psybers** [comment]
This merge has conflicts.

**Review by psybers — CHANGES_REQUESTED** [review]
The merge conflicts.

File: website/validators.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from django.core.exceptions import ValidationError
+
+
+def validate_file_size(value):
+    filesize = value.size
+    
+    if filesize > 4 * 1024 * 1024:
+        raise ValidationError("The maximum file size that can be uploaded is 0")
```
Comment (psybers) [inline website/validators.py]: max is 0?

File: website/views.py [hunk 2]
```diff
@@ -105,7 +105,11 @@ def post(self, request):
                 user.first_name = first_name
                 user.last_name = last_name
                 if len(request.FILES) != 0:
-                    profile.photo = photo
+                    if (photo.size <= 4 * 1024 * 1024):
```
Comment (psybers) [inline website/views.py]: can we have this magic number in only 1 location as a constant?  Perhaps it makes sense to put into the site settings? Maybe also the max image size (500x500) as well?

**Review by psybers — CHANGES_REQUESTED** [review]

File: website/validators.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+from django.core.exceptions import ValidationError
+from django.conf import settings
+
+def validate_file_size(value):
+    filesize = value.size
+    
+    if filesize > settings.MAX_FILE_UPLOAD:
+        raise ValidationError("The maximum file size that can be uploaded is 4MB")
```
Comment (psybers) [inline website/validators.py]: this error message needs to use the constant

File: website/views.py [hunk 4]
```diff
@@ -105,7 +106,11 @@ def post(self, request):
                 user.first_name = first_name
                 user.last_name = last_name
                 if len(request.FILES) != 0:
-                    profile.photo = photo
+                    if (photo.size <= settings.MAX_FILE_ULOAD):
+                        profile.photo = photo
+                    else:
+                        messages.warning(request, ('File size is too big.'))
```
Comment (psybers) [inline website/views.py]: First, is it possible to avoid code duplication and reuse the validator? Second, if not then the error message could be better (use the one from the validator).

File: website/validators.py (hunk 3 above)
Comment (psybers) [reply website/validators.py]: Also maybe reword slightly, e.g. "The maximum file size for profile photos is " + constant

**Review by psybers — CHANGES_REQUESTED** [review]
